src/Storages/StorageDistributed.cpp

```diff
@@ -526,6 +526,10 @@ void StorageDistributed::read(
     const size_t /*max_block_size*/,
     const unsigned /*num_streams*/)
 {
+    // Return directly if no shard to query.
+    if (query_info.cluster->getShardsInfo().empty())
+        return;
+
```

Comment:
```suggestion
```

@amosbird can you remove these lines to avoid conflicts with proper fix - #21579 ?

Reply:
Heh, coding via github is a pleasure :)